examples/defaulter-gen/generators/defaulter.go

```diff
@@ -731,6 +899,85 @@ func (n *callNode) writeCalls(varName string, isVarPointer bool, sw *generator.S
 	}
 }
 
+func (n *callNode) writeDefaulter(varName string, index string, isVarPointer bool, sw *generator.SnippetWriter) {
```

Comment:
Maybe I'm wrong but I suspect this function could be simplified by decomposing the various steps actually involved into separate functions. Let's mark that as a future improvement.

Reply:
I agree. Originally this function was very small but there ended up being quite a number of corner cases to consider.

Reply:
+1, I have a couple thoughts as well, but they don't impact correctness, so follow up should be fine